pinakes/main/approval/models.py

```diff
@@ -184,9 +184,123 @@ class Meta:
             ),
         ]
 
+    def move_internal_sequence(self, delta):
+        """
+        Move current record up or down relative to other records.
+        delta is an integer indictating how many other records should this
+        record be moved over upward (negative) or downward (positive).
+
+        Internal_sequence is a column used to sort the workflows by users.
+        The user only needs to be able to adjust the sequence of the workflow
+        but does not care the actual value of the internal_sequence. When a new
+        workflow is created it is placed to the end of the list with an
+        internal_sequence number much greater than the value from previous tail
+        of the list. When workflow_a is moved to between workflow_b and
+        workflow_c, its internal_sequence is simply adjusted to the average
+        number of b and c unless the average becomes a decimal, in which case
+        all the internal_sequence numbers will get rebalanced with a big
+        interval again.
+        """
+        if delta == 0:
+            return
+
+        if delta > 0:
+            self.internal_sequence = self._internal_sequence_plus(delta)
+        else:
+            self.internal_sequence = self._internal_sequence_minus(-delta)
+
+    def _internal_sequence_plus(self, delta):
+        if delta > Workflow.objects.count():
```

Comment:
Can you please explain the change. 

In previous iterations delta was compared against `math.inf` \ `-math.inf` constants to determine if the record is moved to the last \ first position. Now it is an extra database query and comparison against total count of records in the database.

Reply:
`delta` is expected to be a number to direct the change. When delta is big (out of bound) we can do something special because we know the new sequence will be definitely at the bottom or top. The caller does not need to use a constant to indicate it is bottom or top. For example delta = 100,000 will be effectively the same as using a constant `bottom`. The caller can however pass in `math.inf` to make sure delta is big enough.

This is an internal method, we try to normalize the argument. At the API level we do use constant `bottom` and `top` so the customer is not forced to use math.inf.

Reply:
The `math.inf` constant has a floating point type. Methods in the PR are based on integer types. The floating point type is not used anywhere else except these magic constants. This make the code unclear, hard to follow. Usage of inconsistent data types without a good reason makes this code smell.

Reply:
> delta is expected to be a number to direct the change. When delta is big (out of bound) we can do something special because we know the new sequence will be definitely at the bottom or top.

Please help me to understand the following. If you're comparing the `delta` with Workflow objects count, how does this code handle the situation when let's say the workflow is second to last and the delta is lower then total count.

For example:

Total workflows: 10
Current workflow has the position 9 out of 10.
The `delta` is 5.